### PR TITLE
Stop routing session events via the DAP store

### PR DESCRIPTION
**Pull request by ConradIrwin**
This cleans up a bunch of indirection and will make it easier to
show the session building state in the debugger terminal

Closes #ISSUE

Release Notes:

- N/A

### CODE REVIEW

File: crates/debugger_ui/src/tests/debugger_panel.rs [hunk 1]
```diff
@@ -523,8 +523,8 @@ async fn test_handle_error_run_in_terminal_reverse_request(
         .fake_reverse_request::<RunInTerminal>(RunInTerminalRequestArguments {
             kind: None,
             title: None,
-            cwd: path!("/non-existing/path").into(), // invalid/non-existing path will cause the terminal spawn to fail
-            args: vec![],
+            cwd: "".into(),
```
Comment (Anthony-Eid) [inline crates/debugger_ui/src/tests/debugger_panel.rs:526]: I'm not sure if we should leave this empty because we might have to default to the cwd of the project instead or the  user's HOME directory.

Comment (ConradIrwin) [reply crates/debugger_ui/src/tests/debugger_panel.rs:526]: I don’t think that matters for this test?

Comment (Anthony-Eid) [reply crates/debugger_ui/src/tests/debugger_panel.rs:526]: The name for this test was misleading because it was originally testing that we return an error when we get a run in terminal request with an invalid path. 

Do you have anything against me adding the invalid path check again and creating another test for that edge case?

Comment (ConradIrwin) [reply crates/debugger_ui/src/tests/debugger_panel.rs:526]: The check we had called out to the filesystem synchronously on the main thread (with `try_exists`), so if you do bring it back it needs to use the async methods on the `Arc<dyn Fs>` (or be in the background).

That said, I don't think we should bring it back; because it seems like spawning processes with an invalid `cwd` works just fine.

However I may be missing something. Do you know why the original check was added?
